src/main/java/gregtech/api/recipes/map/MapItemStackNBTIngredient.java

```diff
@@ -37,19 +48,7 @@ public boolean equals(Object obj) {
         }
         if (obj instanceof MapItemStackNBTIngredient) {
             MapItemStackNBTIngredient other = (MapItemStackNBTIngredient) obj;
-            if (this.matcher != null && other.matcher != null) {
-                if (!this.matcher.equals(other.matcher)) {
-                    return false;
-                }
-            }
-            if (this.condition != null && other.condition != null) {
-                if (!this.condition.equals(other.condition)) {
-                    return false;
-                }
-            }
-            //NBT condition is only available on the MapItemStackNBTIngredient created by from the Recipe, so
-            //the evaluate method is called from the comparing MapItemStackNBTIngredient that is on the RecipeMap
-            return ItemStack.areItemsEqual(stack, other.stack) && other.matcher.evaluate(this.stack, other.condition);
+            return other.gtRecipeInput.acceptsStack(this.stack);
```

Comment:
Wouldn't this always fail if the first constructor is used, as `gtRecipeInput` would not be initialized?

Reply:
we will only be matching MapItemStackNBTIngredient made from the itemstacks in the inventory against those made from GTrecipeInputs